downloadtools.ps1

```diff
@@ -0,0 +1,302 @@
+    }
+
+    Write-Output "[Untar-File] Extraction completed. Time taken: $howlong"
+}
+
+function Download-BuildTools
+{
+    # Create a cmd file to update environment variable
+    $path = [Environment]::GetEnvironmentVariable("path").ToLower()
+    $envStream = [System.IO.StreamWriter] "$toolsDir\updatebuildtoolenv.cmd"
+    
+    # TarTool
+    $tarToolExe = "$toolsDir\TarTool.exe"
+    if (!(test-path $tarToolExe))
+    {
+        $url = "http://download-codeplex.sec.s-msft.com/Download/Release?ProjectName=tartool&DownloadId=79064&FileTime=128946542158770000&Build=21031"
+        $output="$toolsDir\TarTool.zip"
+        Download-File $url $output
+        Unzip-File $output $toolsDir
+    }
+    else
+    {
+        Write-Output "$tarToolExe existed already. No download and extraction needed"
+    }
+    
+    if (!($path -like "*$toolsDir*"))
+    {
+        # add toolsdir to path
+        $envStream.WriteLine("set path=$toolsDir;%path%");
+    }
+    
+    # Apache Maven
+    $mvnCmd = "$toolsDir\apache-maven-3.3.3\bin\mvn.cmd"
+    if (!(test-path $mvnCmd))
+    {
+        $url = "http://www.us.apache.org/dist/maven/maven-3/3.3.3/binaries/apache-maven-3.3.3-bin.tar.gz"
+        $output="$toolsDir\apache-maven-3.3.3-bin.tar.gz"
+        Download-File $url $output
+        Untar-File $output $toolsDir
+    }
+    else
+    {
+        Write-Output "$mvnCmd existed already. No download and extraction needed"
+    }
+    
+    $mavenBin = "$toolsDir\apache-maven-3.3.3\bin"
+    if (!($path -like "*$mavenBin*"))
+    {
+        # add maven bin
+        $envStream.WriteLine("set path=$mavenBin\;%path%");
+    }
+    
+    # Nuget Client
+    $nugetExe = "$toolsDir\nuget.exe"
+    if (!(test-path $nugetExe))
+    {
+        $url = "http://dist.nuget.org/win-x86-commandline/latest/nuget.exe"
+        $output=$nugetExe
+        Download-File $url $output
+    }
+    else
+    {
+        Write-Output "$nugetExe existed already. No download and extraction needed"
+    }
+
+    $envStream.close()
+}
+
+function Download-RuntimeDependencies
+{
+    # Create a cmd file to update environment variable
+    $path = [Environment]::GetEnvironmentVariable("path").ToLower()
+    $envStream = [System.IO.StreamWriter] "$toolsDir\updateruntime.cmd"
+
+    # TarTool
+    $tarToolExe = "$toolsDir\TarTool.exe"
+    if (!(test-path $tarToolExe))
+    {
+        $url = "http://download-codeplex.sec.s-msft.com/Download/Release?ProjectName=tartool&DownloadId=79064&FileTime=128946542158770000&Build=21031"
+        $output="$toolsDir\TarTool.zip"
+        Download-File $url $output
+        Unzip-File $output $toolsDir
+    }
+    else
+    {
+        Write-Output "$tarToolExe existed already. No download and extraction needed"
+    }
+    
+    if (!($path -like "*$toolsDir*"))
+    {
+        # add toolsdir to path
+        $envStream.WriteLine("set path=$toolsDir;%path%");
+    }
+
+    # Download Spark binary
+    $S_HOME = "$toolsDir\spark-1.4.1-bin-hadoop2.6"
```

Comment:
Minor nit: you might want to pull out variables for spark-version and hadoop-version then substitute them into the various places in theses strings. That way, only one line change to change version :)